url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/util/http/TestForHttpRequestLine.java

```diff
@@ -0,0 +1,41 @@
+package net.hetimatan.util.http;
+
+import net.hetimatan.io.file.MarkableFileReader;
+import net.hetimatan.io.filen.CashKyoroFile;
+import net.hetimatan.util.http.HttpObject;
+import net.hetimatan.util.http.HttpRequestLine;
+
+import java.io.IOException;
+
+import junit.framework.TestCase;
+
+public class TestForHttpRequestLine extends TestCase {
+
+	public void testHello() {
+
+	}
+
+	public void testEncode001() throws IOException {
+		HttpRequestLine line = new HttpRequestLine(HttpRequestLine.GET, "/announce?a=b&b=c", HttpRequestLine.HTTP11);
+		CashKyoroFile output = new CashKyoroFile(512);
+		line.encode(output.getLastOutput());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *getLastOutput*. Currently, there are execution paths that do not contain closure statements, for example, when HttpRequestLine.encode throws an exception. Ensure the object returned by *getLastOutput()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).